src/diffusers/pipelines/audioldm2/pipeline_audioldm2_tts.py

```diff
@@ -0,0 +1,1002 @@
+                The sequence used as a prompt for the generation.
+            max_new_tokens (`int`):
+                Number of new tokens to generate.
+            model_kwargs (`Dict[str, Any]`, *optional*):
+                Ad hoc parametrization of additional model-specific kwargs that will be forwarded to the `forward`
+                function of the model.
+
+        Return:
+            `inputs_embeds (`torch.FloatTensor` of shape `(batch_size, sequence_length, hidden_size)`):
+                The sequence of generated hidden-states.
+        """
+        max_new_tokens = max_new_tokens if max_new_tokens is not None else self.language_model.config.max_new_tokens
+        for _ in range(max_new_tokens):
+            # prepare model inputs
+            model_inputs = prepare_inputs_for_generation(inputs_embeds, **model_kwargs)
+
+            # forward pass to get next hidden states
+            output = self.language_model(**model_inputs, return_dict=True)
+
+            next_hidden_states = output.last_hidden_state
+
+            # Update the model input
+            inputs_embeds = torch.cat([inputs_embeds, next_hidden_states[:, -1:, :]], dim=1)
+
+            # Update generated hidden states, model inputs, and length for next step
+            model_kwargs = self.language_model._update_model_kwargs_for_generation(output, model_kwargs)
+
+        return inputs_embeds[:, -max_new_tokens:, :]
+
+    def text_to_phoneme(self, transcription):
+        """
+        Preprocess and convert text to phonemes
+        """
+        transcription = re.sub(r"<.*?>", "", transcription)
+        transcription = unidecode(transcription)
+        transcription = transcription.lower()
+        phonemes = phonemize(
+            transcription, language="en-us", backend="espeak", strip=True, preserve_punctuation=True, with_stress=True
+        )
+        phonemes = re.sub(re.compile(r"\s+"), " ", phonemes)
+        phonemes += "⚠"  # The last token in offical audioldm2
+
+        return phonemes
+
+    def encode_prompt(
+        self,
+        prompt,
+        transcription,
+        device,
+        num_waveforms_per_prompt,
+        do_classifier_free_guidance,
+        negative_prompt=None,
+        prompt_embeds: Optional[torch.FloatTensor] = None,
+        negative_prompt_embeds: Optional[torch.FloatTensor] = None,
+        generated_prompt_embeds: Optional[torch.FloatTensor] = None,
+        negative_generated_prompt_embeds: Optional[torch.FloatTensor] = None,
+        attention_mask: Optional[torch.LongTensor] = None,
+        negative_attention_mask: Optional[torch.LongTensor] = None,
+        max_new_tokens: int = None,
+    ):
+        r"""
+        Encodes the prompt into text encoder hidden states.
+
+        Args:
+            prompt (`str` or `List[str]`, *optional*):
+                prompt to be encoded
+            transcription (`str`):
```

Comment:
Could also be a list of strings no? If we have a list of prompts, then we'd have one transcription for each prompt?